common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -34,6 +34,27 @@
  * Utility class for collation-aware UTF8String operations.
  */
 public class CollationAwareUTF8String {
+
+  public static boolean lowercaseMatchFrom(final UTF8String l, final UTF8String r, int pos) {
+    if (pos < 0) return false;
+    for (int len = 0; len <= l.numChars() - pos; len++) {
```

Comment:
```suggestion
    for (int len = 0; len <= l.numChars() - pos; ++len) {
```

Reply:
actually, I think "lowercaseMatchFrom" should be able to match empty substring of `l` (len=0) to `r` (if `r` is an empty string itself)

otherwise, I would need to add a special case before the loop to cover this behaviour - I think there's no need to do that, since the first iteration of this loop will cover that scenario